app/logic/searchUsers.py

```diff
@@ -9,17 +11,35 @@ def searchUsers(query):
 
     firstName = splitSearch[0] + "%"
     lastName = " ".join(splitSearch[1:]) +"%"
-
+    results = None
+    searchId = None
     if len(splitSearch) == 1: #search for first or last name
-        results = User.select().where(User.isStudent & (User.firstName ** firstName | User.lastName ** firstName))
+        if searchGroup != "outsideParticipant":
+            results = User.select().where(User.isStudent & (User.firstName ** firstName | User.lastName ** firstName))
+        else:
+            results = OutsideParticipant.select().where(OutsideParticipant.firstName ** firstName | OutsideParticipant.lastName ** firstName)
+
         for participant in results:
-            if participant not in resultsDict:
-                resultsDict[participant.username]=model_to_dict(participant)
+            if searchGroup != "outsideParticipant":
+                searchId = participant.username
+                if participant not in resultsDict:
+                    resultsDict[participant.username]=model_to_dict(participant)
+            else:
+                searchId = participant.email
+                if participant not in resultsDict:
+                    resultsDict[participant.email]=model_to_dict(participant)
```

Comment:
The only thing actually different in these two blocks is the value you are using for a search id. Refactor so the if/else is around the smallest code, and the common code is outside